src/sharepoint/order/sharepoint-order.controller.js

```diff
@@ -103,98 +131,74 @@ angular
-    checkSharepointActivation(account) {
-      if (account.activateSharepoint) {
-        this.accountsToActivate.push(account.primaryEmailAddress);
-      } else {
-        this.accountsToActivate
-          .splice(this.accountsToActivate.indexOf(account.primaryEmailAddress), 1);
+    onAssociateChange(associateExchange) {
+      if (!associateExchange) {
+        // we need to reset selected accounts if we leave the association interface
+        this.accountsToAssociate = [];
       }
     }
 
-    getExchangeAccountsUrl() {
-      return `#/configuration/exchange_hosted/${this.associatedExchange.organization}/${this.associatedExchange.name}?tab=ACCOUNTS`;
-    }
-
-    static hasInexistantAccount(activateSharepointForm) {
-      return activateSharepointForm.primaryEmailAddressField.$invalid
-        && !_.isEmpty(activateSharepointForm.primaryEmailAddressField.$viewValue);
-    }
-
-    changeStandAloneQuantity() {
-      if (parseInt(this.standAloneQuantity, 10) >= 1
-        && parseInt(this.standAloneQuantity, 10) <= 30) {
-        this.standAloneQuantity = parseInt(this.standAloneQuantity, 10);
-      }
+    refreshAccounts(exchange) {
+      this.associatedExchange = exchange;
+      this.ouiDatagridService.refresh('exchangeAccountsDatagrid');
     }
 
-    decrement() {
-      if (this.standAloneQuantity > 1) {
-        this.standAloneQuantity -= 1;
-      }
+    onAccountsSelected(accounts) {
+      this.accountsToAssociate = accounts.map(({ primaryEmailAddress }) => primaryEmailAddress);
     }
 
-    increment() {
-      if (this.standAloneQuantity < 30) {
-        this.standAloneQuantity += 1;
-      }
+    hasSelectedAccounts() {
+      return this.accountsToAssociate.length > 0;
     }
 
     getSharepointOrderUrl() {
       if (this.associateExchange) {
-        if (_.has(this.associatedExchange, 'name') && this.accountsToActivate.length >= 1) {
-          return this.sharepointService.getSharepointOrderUrl(
+        if (_.has(this.associatedExchange, 'name') && this.accountsToAssociate.length >= 1) {
+          return this.Sharepoint.getSharepointOrderUrl(
             this.associatedExchange.name,
-            this.accountsToActivate,
+            this.accountsToAssociate,
           );
         }
         return '';
       }
-      if (!_.isNull(this.standAloneQuantity)
-        && parseInt(this.standAloneQuantity, 10) >= 1
-        && parseInt(this.standAloneQuantity, 10) <= 30) {
-        return this.sharepointService
-          .getSharepointStandaloneOrderUrl(parseInt(this.standAloneQuantity, 10));
+
+      const quantity = parseInt(this.standAloneQuantity, 10);
+      if (quantity >= 1 && quantity <= 30) {
+        return this.isReseller
+          ? this.Sharepoint.getSharepointProviderOrderUrl(quantity)
+          : this.Sharepoint.getSharepointStandaloneOrderUrl(quantity);
       }
 
       return '';
     }
+
+    goToSharepointOrder() {
+      this.$window.open(this.getSharepointOrderUrl(), '_blank', 'noopener');
```

Comment:
noopener is valid: https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Window_functionality_features
The fact that on some browsers it opens an new window instead of a new tab is user preference: https://stackoverflow.com/questions/4907843/open-a-url-in-a-new-tab-and-not-a-new-window-using-javascript
Nothing can be done against this.

Reply:
You're right, noopener is valid (wrong documentation page).

But the difference is between

```js
window.open('https://www.ovh.com/', '_blank', 'noopener');
```
that opens a new window

and 

```js
window.open('https://www.ovh.com/', '_blank');
```
that opens a new tab.

I think you want to reproduce the behaviour of :

```html 
<a href="https://www.ovh.com/" target="_blank" rel="noopener">OVH</a>
```

in javascript 
```js
window.open('https://www.ovh.com/', '_blank', 'noopener');
```

but the result is not the same... 



=> If you want it you should do something like this : 

```js
const newWindow = window.open();
newWindow.opener = null;
newWindow.location = 'https://www.ovh.com/';
```